### PR TITLE
DOC/STY: docstrings, classes in test_custom and test_methods_general

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses #860, #565 (partial)

Applies new style to docstrings and code in:
- test_custom.py
- test_methods_general.py

Style changes include
- New docstring standards
- Declare classes as objects
- `return` statements in unit tests

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update

# How Has This Been Tested?

via `flake-docstrings` and `hacking`:
```
flake8 --select=D,E,F,H,W --count --statistics
```

**Test Configuration**:
* Operating system: Mac OS X 10.15.7
* Version number: Python 3.8.2
* flake8, flake-docstrings, hacking

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes (part of series already documented)

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: pysat/tests/test_custom.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
+"""Unit Tests for the `custom_attach` methods."""
```
Comment (aburrell) [inline pysat/tests/test_custom.py:1]:
```suggestion
"""Unit tests for the `custom_attach` methods."""
```

File: pysat/tests/test_custom.py [hunk 2]
```diff
@@ -33,71 +35,82 @@ def mult_data(inst, mult, dkey="mlt"):
...
                                     at_pos=3)
         self.out = self.log_capture.getvalue()
 
         assert self.out.find(
             "unknown position specified, including function at end") >= 0
+        return
+
 
+class TestBasics(object):
+    """Unit tests for `pysat.instrument.custom_attach` with a pandas inst."""
```
Comment (aburrell) [inline pysat/tests/test_custom.py:72]:
```suggestion
    """Unit tests for `pysat.instrument.custom_attach` with pandas data."""
```

File: pysat/tests/test_methods_general.py [hunk 3]
```diff
@@ -1,3 +1,5 @@
+"""Unit Tests for the general instrument methods."""
```
Comment (aburrell) [inline pysat/tests/test_methods_general.py:1]:
```suggestion
"""Unit tests for the general instrument methods."""
```

File: pysat/tests/test_methods_general.py [hunk 4]
```diff
@@ -7,78 +9,101 @@
 from pysat.instruments.methods import general as gen
 
 
-class TestGenMethods():
+class TestListFiles(object):
+    """Unit Tests for `pysat.instrument.methods.general.list_files`."""
```
Comment (aburrell) [inline pysat/tests/test_methods_general.py:13]:
```suggestion
    """Unit tests for `pysat.instrument.methods.general.list_files`."""
```

File: pysat/tests/test_methods_general.py [hunk 5]
```diff
@@ -7,78 +9,101 @@
...
+
         in_time = pds.DateOffset(**{time_kwarg: time_val})
         check_daily = gen.is_daily_file_cadence(in_time)
 
         assert check_daily == is_daily
+        return
+
 
+class TestRemoveLeadText(object):
+    """Unit Tests for `pysat.instrument.methods.general.remove_leading_text`."""
```
Comment (aburrell) [inline pysat/tests/test_methods_general.py:77]:
```suggestion
    """Unit tests for `pysat.instrument.methods.general.remove_leading_text`."""
```

File: pysat/tests/test_methods_general.py [hunk 6]
```diff
@@ -117,23 +148,37 @@ def test_remove_names_w_target_list(self):
         # Check prepended text removed from metadata
         assert '_Blurp' in self.testInst.meta.keys()
         assert 'loop' in self.testInst.meta.keys()
+        return
 
 
 class TestRemoveLeadTextXarray(TestRemoveLeadText):
+    """Unit Tests for `pysat.instrument.methods.general.remove_leading_text`.
```
Comment (aburrell) [inline pysat/tests/test_methods_general.py:155]:
```suggestion
    """Unit tests for `pysat.instrument.methods.general.remove_leading_text`.
```

File: pysat/tests/test_methods_general.py [hunk 7]
```diff
@@ -155,21 +203,28 @@ def test_remove_2D_names_w_target_list(self):
         # Check prepended text removed from metadata
         assert '_profiles' in self.testInst.meta.keys()
         assert 'ages' in self.testInst.meta.keys()
+        return
+
 
+class TestLoadCSVData(object):
+    """Unit Tests for `pysat.instrument.methods.general.load_csv_data`."""
```
Comment (aburrell) [inline pysat/tests/test_methods_general.py:210]:
```suggestion
    """Unit tests for `pysat.instrument.methods.general.load_csv_data`."""
```

File: pysat/tests/test_custom.py [hunk 8]
```diff
@@ -114,10 +127,11 @@ def test_basic_str_w_function(self):
         assert self.out.find('mult_data') > 0
         assert self.out.find('Args') > 0
         assert self.out.find('Kwargs') > 0
+        return
 
     def test_single_modifying_custom_function_error(self):
-        """Test for error when custom function loaded as modify returns a value
-        """
+        """Test for error if 'modify' custom function returns a value."""
```
Comment (rstoneback) [inline pysat/tests/test_custom.py:133]: 'modify' doesn't exist anymore. There are only custom functions now.
```suggestion
        """Test for error if custom function returns a value."""
```

File: pysat/tests/test_custom.py [hunk 9]
```diff
@@ -245,23 +266,26 @@ def setup(self):
...
         self.testConst.custom_attach(mult_data, args=self.custom_args)
         self.out = self.testConst.__repr__()
         assert isinstance(self.out, str)
         assert self.out.find("'function'") >= 0
+        return
 
     def test_single_modifying_custom_function_error(self):
-        """Test for error when custom function loaded as modify returns a value
-        """
+        """Test for error when 'modify' custom function returns a value."""
```
Comment (rstoneback) [inline pysat/tests/test_custom.py:287]:
```suggestion
        """Test for error when custom function returns a value."""
```

File: pysat/tests/test_custom.py [hunk 10]
```diff
@@ -245,23 +266,26 @@ def setup(self):
...
-        """Test __repr__ with a custom method"""
+        """Test __repr__ with a custom method."""
 
         self.testConst.custom_attach(mult_data, args=self.custom_args)
         self.out = self.testConst.__repr__()
         assert isinstance(self.out, str)
         assert self.out.find("'function'") >= 0
+        return
 
     def test_single_modifying_custom_function_error(self):
```
Comment (rstoneback) [inline pysat/tests/test_custom.py:286]:
```suggestion
    def test_single_custom_function_error(self):
```

File: pysat/tests/test_custom.py [hunk 11]
```diff
@@ -114,10 +127,11 @@ def test_basic_str_w_function(self):
         assert self.out.find('mult_data') > 0
         assert self.out.find('Args') > 0
         assert self.out.find('Kwargs') > 0
+        return
 
     def test_single_modifying_custom_function_error(self):
```
Comment (jklenzing) [inline pysat/tests/test_custom.py:132]:
```suggestion
    def test_single_custom_function_error(self):
```

File: pysat/tests/test_custom.py [hunk 12]
```diff
@@ -245,23 +266,26 @@ def setup(self):
...
-        """ Runs after every method to clean up previous testing
-        """
+        """Clean up the unit test environment after each method."""
+
         del self.testConst, self.load_date, self.custom_args
+        return
 
     def test_basic_repr(self):
-        """Test __repr__ with a custom method"""
+        """Test __repr__ with a custom method."""
```
Comment (aburrell) [inline pysat/tests/test_custom.py:278]:
```suggestion
        """Test `__repr__` with a custom method."""
```

File: pysat/tests/test_methods_general.py [hunk 13]
```diff
@@ -7,78 +9,101 @@
...
         self.kwargs[bad_key] = bad_val
         with pytest.raises(ValueError) as excinfo:
             gen.list_files(**self.kwargs)
         assert str(excinfo.value).find(err_msg) >= 0
+        return
 
 
-class TestFileCadence():
+class TestFileCadence(object):
+    """Unit Tests for `instruments.methods.general.is_daily_file_cadence`."""
```
Comment (aburrell) [inline pysat/tests/test_methods_general.py:46]:
```suggestion
    """Unit tests for `instruments.methods.general.is_daily_file_cadence`."""
```

**Review by aburrell — APPROVED** [review]

**Pull request merged**